flang/lib/Lower/ConvertVariable.cpp

```diff
@@ -1804,9 +1804,38 @@ void Fortran::lower::mapCallInterfaceSymbols(
       instantiateVariable(converter, var, symMap, storeMap);
     } else {
       const Fortran::semantics::Symbol &sym = var.getSymbol();
-      // Get the argument for the dummy argument symbols of the current call.
-      if (Fortran::semantics::IsDummy(sym) && sym.owner() == result.owner())
+      const auto *hostDetails =
+          sym.detailsIf<Fortran::semantics::HostAssocDetails>();
+      if (hostDetails && !var.isModuleVariable()) {
+        // The callee is an internal procedure `A` whose result properties
+        // depend on host variables. There caller may be the host, or another
+        // internal procedure `B` contained in the same host.  In the first
+        // case, the host symbol is obviously mapped, in the second case, it
+        // must also be mapped because
+        // HostAssociations::internalProcedureBindings that was called when
+        // lowering `B` did map all host symbols of captured variables to the
+        // variable from the tuple argument, whether the host symbol is actually
```

Comment:
s/variable from the tuple argument/the tuple argument containing the composite of all host associated variables/